### PR TITLE
MOAB driver changes for eamxx 

### PR DESCRIPTION
**Pull request by iulian787**
EAMXX uses a different implementation of the atmosphere model, compared to EAM, and moab driver needs to be extended to support a coupled case

case tested is --compset F2010-SCREAMv1-MPASSI --res ne4pg2_oQU480  --driver moab
this case uses mpas sea ice, CICE is not yet supported by MOAB

the implementation follows similar strategy compared to the rest of the models
Atmosphere mesh in moab has 4 different instances, similar to EAM,  
1) corresponding to coarse spectral mesh ( size 6*ne*ne), 
2) fine quad mesh (corresponding to GLL quad mesh, size 6*ne*ne* (np-1)* (np-1) )
3)  pg2 FV type mesh (size 6*ne*ne* pg2 * pg2) 
4)  point cloud corresponding to mct simplified vertex grid.

On the coupler side, we still use just the pg2 FV type mesh, used for intersection and coupling with other models

-----

Major difference compared to eam is the fields import and export. 
In eam and eamxx, atm_import and atm_export for mct is using the mct av vectors directly,  for host ( and duplicated for device )

MOAB needs extra buffers, but those buffers are transposed compared to mct driver, because moab aligns data differently

We currently keep both mct and moab buffers, because we are running both drivers for comparison purposes; (like for all components)
At some point, mct scaffolding will be removed from all components, so data will not be duplicated anymore

major difference between fields buffers is the fact that for moab buffers the column index varies faster in the 2d array (while in mct, fields index varies faster).  need to be careful at parallel_for used to copy data back an forth

### CODE REVIEW

**Comment by rljacob** [comment]
Note:  to test this on Sandia machines, MOAB will need to be installed.  https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/4019814452/Installing+MOAB+and+TempestRemap+for+E3SM
Its already installed on Perlmutter and Chrysalis.

File: components/eamxx/src/mct_coupling/atm_comp_mct.F90
Comment (bartgol) [inline components/eamxx/src/mct_coupling/atm_comp_mct.F90:500]: Can you add some comments at the top of this function, explaining what it does and when it's called?

**Review by bartgol — COMMENTED** [review]
Looks good, but I'll let @tcclevenger decide, since he worked more than me (and more recently) on SC stuff. I have only one minor request.

**Comment by rljacob** [comment]
@bartgol @AaronDonahue or @tcclevenger can one of you approve this PR?  Approval just means you're ok with changes to code in eamxx/src.  They should have no effect on your use of driver-mct.

**Review by tcclevenger — APPROVED** [review]
SC stuff looks good to me!

**Comment by tcclevenger** [comment]
Feel free to set AUTOMERGE to get this in.

File: components/eamxx/src/control/atmosphere_surface_coupling_importer.cpp
Comment (bartgol) [inline components/eamxx/src/control/atmosphere_surface_coupling_importer.cpp:267]: @iulian787 Is something supposed to happen here before we merge?

Comment (iulian787) [reply components/eamxx/src/control/atmosphere_surface_coupling_importer.cpp:267]: @bartgol the case I tested (--compset F2010-SCREAMv1-MPASSI --res ne4pg2_oQU480 ) does not have iop_srf_prop flag to true, so the code does not enter this method;
I left here a mark that when we will have a case with iop_srf_prop true, MOAB will have to do something else; I am not sure yet. 
I could take it out. 

I see some test cases here
components/eamxx/cime_config/testdefs/testmods_dirs/scream/dpxx/arm97/shell_commands
components/eamxx/cime_config/testdefs/testmods_dirs/scream/dpxx/dycomsrf01/shell_commands

for which iop_srf_prop is true; 

@rljacob should I try these cases  ?  I am not even sure how to run them

Comment (rljacob) [reply components/eamxx/src/control/atmosphere_surface_coupling_importer.cpp:267]: No. We'll handle the single column cases in another PR.

Comment (bartgol) [reply components/eamxx/src/control/atmosphere_surface_coupling_importer.cpp:267]: Ok, thanks!

**Comment by AaronDonahue** [comment]
@rljacob and @bartgol can you confirm that your comments have been addressed on this PR?  I'll add the AUTOMERGE label once you have.  Broad question, with `weaver` coming back online can we rely on the AT or do we need manually test on perlmutter?  Follow-up, re: Rob's comment, is MOAB installed on `weaver` and `mappy`?

**Comment by rljacob** [comment]
Yes they've been addressed.  You don't need to install MOAB on weaver or mappy for this PR.

**Comment by bartgol** [comment]
FYI, I recently changed the team name "scream team" to "scream-team", to avoid issues in bash regarding spaces in some other script. I had forgotten to update the AT config files, so it refused to consider the approval valid. Now it should work.

**Comment by iulian787** [comment]
I compiled MOAB on perlmutter using "gnugpu" environment,
I tested both pm-cpu and pm-gpu runs with moab driver, and they worked fine

 I added MOAB_ROOT to the machine/config_machines.xml on this branch, for pm-gpu machine

**Review by bartgol — APPROVED** [review]

**Comment by bartgol** [comment]
@iulian787 if you are happy with the PR, feel free to merge.

**Comment by rljacob** [comment]
Using the green button?

**Comment by bartgol** [comment]
> Using the green button?

Yes. I can of course do it too, but since it's not my PR, I don't want to merge in case @iulian787 was thinking to add some other commit.

**Pull request merged**
